src/contacts/model/ContactModel.ts

```diff
@@ -114,6 +120,49 @@ export class ContactModelImpl implements ContactModel {
 		)
 		return loadedContacts.flat()
 	}
+
+	async searchForContactLists(query: string): Promise<ContactListInfo[]> {
+		if (!this.loginController.isFullyLoggedIn()) {
+			throw new LoginIncompleteError("cannot search for contact lists as online login is not completed")
+		}
+
+		const contactLists = await this.getLoadedContactListInfos()
+
+		return contactLists.filter((contactList) => contactList.name.toLowerCase().includes(query))
+	}
+
+	private async loadContactLists() {
+		const userController = this.loginController.getUserController()
+		const contactListMemberships = userController.getContactListMemberships()
+		this.contactListInfo = (
+			await promiseMap(
+				await promiseMap(contactListMemberships, (rlm: GroupMembership) => this.entityClient.load(GroupInfoTypeRef, rlm.groupInfo)),
+				// we might still have a membership for a short time when the group root is already deleted
+				(groupInfo) => this.getContactListInfo(groupInfo).catch(ofClass(NotFoundError, () => null)),
+			)
+		).filter(isNotNull)
+	}
+
+	private async getContactListInfo(groupInfo: GroupInfo): Promise<ContactListInfo> {
+		const groupRoot = await this.entityClient.load(ContactListGroupRootTypeRef, groupInfo.group)
+
+		const { getSharedGroupName } = await import("../../sharing/GroupUtils.js")
+
+		return {
+			name: getSharedGroupName(groupInfo, this.loginController.getUserController(), true),
+			groupInfo,
+			groupRoot,
+		}
+	}
+
+	private readonly entityEventsReceived: EntityEventsListener = async (updates: ReadonlyArray<EntityUpdateData>): Promise<void> => {
+		for (const update of updates) {
+			if (isUpdateForTypeRef(UserTypeRef, update) && isSameId(this.loginController.getUserController().userId, update.instanceId)) {
+				await this.loadContactLists()
+			}
+			// FIXME we need to either redraw or signal somehow that the contact list has changed
```

Comment:
fixme. It's also not clear yet _if_ a contact list has changed here, right? we only checked if the user changed